lib/media/drm_engine.js

```diff
@@ -1454,7 +1454,8 @@ shaka.media.DrmEngine = class {
     // Some CDMs do not have sub-second time resolution, so the key status may
     // fire with hundreds of milliseconds left until the stated expiration time.
     const msUntilExpiration = session.expiration - Date.now();
-    if (msUntilExpiration < 0 || (hasExpiredKeys && msUntilExpiration < 1000)) {
+    if ((msUntilExpiration < 0 && this.config_.checkDrmExpirationStatus) ||
```

Comment:
The naming of this config flag is a bit confusing like this. It says "check status", but in fact it enables checking of the expiration time only.
I would name it "ignoreDrmExpirationStatus" then, because if this flag is true, it ignores the second part of the "if" condition about the status.